### PR TITLE
fix: the build process may timeout when the message is too long

### PR DESCRIPTION
**Pull request by piglei**
fix: the build process may timeout when the message is too long, changes include:

1. truncate the log message if it's too long (>4k) and only match the first 1k characters when doing step pattern matching
2. improve the step pattern by replacing "\s+" with "\s", the latter is much faster when searching a string with many blank characters
3. reduce the number of patterns by introducing re.IGNORECASE.

### CODE REVIEW

File: apiserver/paasng/paasng/platform/engine/utils/output.py [hunk 1]
```diff
@@ -208,3 +201,19 @@ def get_default_stream(deployment: Deployment) -> RedisChannelStream:
...
+
+    Sometimes the message was generated by program and can be very long(>10k), writing
+    it directly to stream will cause performance issue and took too much storage, so
+    we need to truncate it.
+    """
+    max_length = 4096
+    if len(message) > max_length:
+        message = message[:max_length] + "...(content too long, truncated)"
+
+    message = message.replace("\x1b[1G", "")
```
Comment (jamesgetx) [inline apiserver/paasng/paasng/platform/engine/utils/output.py:218]: Nit: `return message.replace("\x1b[1G", "")`

**Review by jamesgetx — APPROVED** [review]

File: apiserver/paasng/paasng/platform/engine/utils/output.py [hunk 2]
```diff
@@ -208,3 +201,19 @@ def get_default_stream(deployment: Deployment) -> RedisChannelStream:
...
+    """Sanitize the given message, remove bad characters and truncate it if it's too
+    long.
+
+    Sometimes the message was generated by program and can be very long(>10k), writing
+    it directly to stream will cause performance issue and took too much storage, so
+    we need to truncate it.
+    """
+    max_length = 4096
+    if len(message) > max_length:
+        message = message[:max_length] + "...(content too long, truncated)"
```
Comment (jamesgetx) [inline apiserver/paasng/paasng/platform/engine/utils/output.py:216]: 可以考虑加上截断的长度，明晰一些
```
message = f"{message[:max_length]}...(content too long, truncated to a length of {max_length})"
```

File: apiserver/paasng/paasng/platform/engine/utils/output.py (hunk 1 above)
Comment (piglei) [reply apiserver/paasng/paasng/platform/engine/utils/output.py:218]: 较细微，暂不调整。

File: apiserver/paasng/paasng/platform/engine/utils/output.py (hunk 2 above)
Comment (piglei) [reply apiserver/paasng/paasng/platform/engine/utils/output.py:216]: 一开始，确实是准备在尾部展示精确的最大长度 `...(content longer than 4096 characters was truncated)`，但具体测试后，发现这段缩略说明有点偏长了，不是很好看……

所以简化成了现在的样子。考虑到绝大多数用户根本不会触发到这个限制，即使触发，具体的最大字符数量本身也不算重要，因此先使用目前的方案。

**Pull request merged**